testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_replication_eventing.py

```diff
@@ -0,0 +1,511 @@
+#     db = params_from_base_test_setup["db"]
+#     cbl_db = params_from_base_test_setup["source_db"]
+#     sync_gateway_version = params_from_base_test_setup["sync_gateway_version"]
+#
+#     if sync_gateway_version < "2.5.0":
+#         pytest.skip('This test cannnot run with sg version below 2.5')
+#     channels = ["ABC"]
+#     username = "autotest"
+#     password = "password"
+#
+#     # Create CBL database
+#     sg_client = MobileRestClient()
+#
+#     # Reset cluster to ensure no data in system
+#     c = cluster.Cluster(config=cluster_config)
+#     c.reset(sg_config_path=sg_config)
+#
+#     # 1. Creating Docs in SG
+#
+#     sg_client.create_user(url=sg_admin_url, db=sg_db, name=username, password=password, channels=channels)
+#     cookie, session = sg_client.create_session(url=sg_admin_url, db=sg_db, name=username)
+#     auth_session = cookie, session
+#     sync_cookie = "{}={}".format(cookie, session)
+#     session_header = {"Cookie": sync_cookie}
+#     sg_docs = sg_client.add_docs(url=sg_url, db=sg_db, number=num_of_docs, id_prefix="sg_docs", channels=channels,
+#                                  auth=auth_session)
+#     sg_doc_ids = [doc["id"] for doc in sg_docs]
+#
+#     # Replicating docs from SG
+#     # Adding Listener for replicator
+#     replicator = Replication(base_url)
+#     repl_config = replicator.configure(source_db=cbl_db,
+#                                        target_url=sg_blip_url,
+#                                        continuous=False,
+#                                        headers=session_header,
+#                                        replication_type="pull")
+#     repl_conflict = replicator.create(repl_config)
+#     repl_error_change_listener = replicator.addReplicatorEventChangeListener(repl_conflict)
+#
+#     # 2. Starting Replication and disabling user during replication to create pull error
+#     replicator.start(repl_conflict)
+#     sg_client.update_user(url=sg_admin_url, db=sg_db, name=username, disabled=True)
+#     try:
+#         replicator.wait_until_replicator_idle(repl_conflict)
+#     except Exception, err:
+#         log_info("Error occured during replication - {}".format(str(err)))
+#
+#     # 3. verifying the error in the events collected
+#     doc_error_repl_event_changes = replicator.getReplicatorEventChanges(repl_error_change_listener)
+#     replicator.removeReplicatorEventListener(repl_conflict, repl_error_change_listener)
+#     replicator.stop(repl_conflict)
+#     event_dict = get_event_changes(doc_error_repl_event_changes)
+#     cbl_doc_ids = db.getDocIds(database=cbl_db)
+#     cbl_docs = db.getDocuments(database=cbl_db, ids=cbl_doc_ids)
+#     assert num_of_docs == len(cbl_docs)
+#     assert num_of_docs == len(event_dict)
+
+
+@pytest.mark.sanity
+@pytest.mark.listener
+@pytest.mark.replication
+@pytest.mark.parametrize("num_of_docs", [
+    10,
+    100,
+])
+def test_replication_access_revoke_event(params_from_base_test_setup, num_of_docs):
```

Comment:
looks like document removal has test something similiar to this. Can I use add few verifications to same test, so that we do not need to duplicate with channel access removal?

Reply:
Please do. You can change it in your branch or make your commit here